[docs] Refactor "Selectors" (#4799)
DevExpress/testcafe: docs/articles/templates/selectors/filter-dom-by-predicates.md

> Functions that search for elements through the DOM tree allow you to filter the matched set by a `filterFn` predicate.

Reviewer: ```suggestion
Functions that search for elements in the DOM tree allow you to use a `filterFn` predicate to filter the matched set.
```

> The `filterFn` predicate is executed on the client. It takes the following parameters.

Reviewer 2: `executed on the client-side` is probably a more standard way to put it

> The `filterFn` predicate is executed on the client side. It takes the following parameters.

Reviewer: ```suggestion
The `filterFn` predicate is executed on the client side and accepts the following parameters:
```

Parameter | Description
------ | -----
`node`  | The current matching node.

> `idx` |  A zero-based index of `node` among other matching nodes.

Reviewer: ```suggestion
`idx` |  A matching node's zero-based index.
```
